src/RedirectMiddleware.php

```diff
@@ -108,23 +110,27 @@ public function checkRedirect(
         if (!empty($options['allow_redirects']['track_redirects'])) {
             return $this->withTracking(
                 $promise,
-                (string) $nextRequest->getUri()
+                (string) $nextRequest->getUri(),
+                (string) $response->getStatusCode()
```

Comment:
Sorry about that, I guess I was a little confused initially. Totally get your point now tho.

Update PR incoming....